File: benchmark/js/fib.js

function fib (x) {
    if (x === 0)
        return 0
    
    
    if (x < 3)
        return 1
    

    return fib(x-1) + fib(x-2)
}

if (process.argv.length < 3) {
    console.error(`${process.argv.slice(0, 2).join(' ')}: Missing argument\n`)
    process.exit(1)
}

let x = parseInt(process.argv[2])
console.log(`fib(${x}) = ${fib(x)}`)

File: benchmark/c/fib.c
#include <stdio.h>
#include <stdlib.h>

unsigned long long fib(const unsigned long long x)
{
    if (x == 0)
        return 0;
    if (x < 3)
        return 1;

    return fib(x - 1) + fib(x - 2);
}

int main(const int argc, const char **argv)
{
    if (argc < 2)
    {
        fprintf(stderr, "%s: missing argument\n", argv[0]);
        exit(1);
    }

    const int x = atoi(argv[1]);

    printf("fib(%d) = %llu\n", x, fib(x));
    return 0;
}